pkg/session_test.go

```diff
@@ -571,3 +573,116 @@ func TestPeerBWithAudioAndVideoWhenPeerAHasAudioOnly(t *testing.T) {
 	peerB.Close()
 	remoteB.Close()
 }
+
+func TestSession_MultiPeerSessionWithTransportNack(t *testing.T) {
+	report := test.CheckRoutines(t)
+	defer report()
+
+	me := webrtc.MediaEngine{}
+	codec := webrtc.NewRTPVP8CodecExt(webrtc.DefaultPayloadTypeVP8, 9000, rtcpfb, "")
+	me.RegisterCodec(codec)
+	api := webrtc.NewAPI(webrtc.WithMediaEngine(me))
+
+	session := NewSession("session")
+	remoteA, remoteB, err := newPair(webrtc.Configuration{}, api)
+	assert.NoError(t, err)
+
+	//// Add a pub track for remote A
+	trackA, err := remoteA.NewTrack(webrtc.DefaultPayloadTypeVP8, rand.Uint32(), "video", "pion")
+	assert.NoError(t, err)
+	_, err = remoteA.AddTrack(trackA)
+	assert.NoError(t, err)
+
+	// Add a datachannel for remote A
+	_, err = remoteA.CreateDataChannel("foo", nil)
+	assert.NoError(t, err)
+
+	//// Add a pub track for remote A
+	trackB, err := remoteB.NewTrack(webrtc.DefaultPayloadTypeVP8, rand.Uint32(), "video", "pion")
+	assert.NoError(t, err)
+	_, err = remoteB.AddTrack(trackB)
+	assert.NoError(t, err)
+
+	// Add a datachannel for remote A
+	_, err = remoteB.CreateDataChannel("bar", nil)
+	assert.NoError(t, err)
+
+	peerA, err := signalPeer(session, remoteA)
+	assert.NoError(t, err)
+	session.AddTransport(peerA)
+
+	pkt := rtcp.TransportLayerNack{
+		SenderSSRC: trackB.SSRC(),
+		MediaSSRC:  trackB.SSRC(),
+		Nacks:      nil,
+	}
+
+	onTrackReceivedCtx, onTrackReceivedFunc := context.WithCancel(context.Background())
+
+	peerA.OnTrack(func(track *webrtc.Track, receiver *webrtc.RTPReceiver) {
+		ctx := context.Background()
+		videoReceiver := NewWebRTCVideoReceiver(ctx, WebRTCVideoReceiverConfig{
```

Comment:
Instead of creating a Receiver here, we should call `sender.ReadRTCP() on track B's sender`

So on line 603, get the sender with

```
sender, err = remoteB.AddTrack(trackB)
```

and then here you can read the RTCP from it